components/add_users_to_group_modal/add_users_to_group_modal.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {localizeMessage} from 'utils/utils.jsx';
+import {Group} from 'mattermost-redux/types/groups';
+
+import 'components/user_groups_modal/user_groups_modal.scss';
+import {ModalData} from 'types/actions';
+import AddUserToGroupMultiSelect from 'components/add_user_to_group_multiselect';
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+
+export type Props = {
+    onExited: () => void;
+    groupId: string;
+    group: Group;
+    backButtonCallback: () => void;
+    actions: {
+        addUsersToGroup: (groupId: string, userIds: string[]) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+const AddUsersToGroupModal = (props: Props) => {
+    const [show, setShow] = useState(true);
+    const [usersToAdd, setUsersToAdd] = useState<UserProfile[]>([]);
+    const [showUnknownError, setShowUnknownError] = useState(false);
+
+    const doHide = () => {
+        setShow(false);
+    };
+
+    const isSaveEnabled = () => {
+        return usersToAdd.length > 0;
+    };
+
+    const addUserCallback = (users: UserProfile[]): void => {
+        setUsersToAdd(users);
+    };
+
+    const deleteUserCallback = (users: UserProfile[]): void => {
+        setUsersToAdd(users);
+    };
+
+    const addUsersToGroup = async (users?: UserProfile[]) => {
```

Comment:
All this functions are regenerated in every single render, triggering unneeded re-renders in the underneath components. Not an stopper, but something too fix sooner than later. I would use `useCallback` in most of this functions.